src/test/scala/com/fulcrumgenomics/umi/DuplexConsensusCallingIntegrationTest.scala

```diff
@@ -0,0 +1,211 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+
+package com.fulcrumgenomics.umi
+
+import com.fulcrumgenomics.bam.api.{SamOrder, SamRecord}
+import com.fulcrumgenomics.testing.SamBuilder.{Minus, Plus, Strand}
+import com.fulcrumgenomics.testing.{ReferenceSetBuilder, SamBuilder, UnitSpec}
+import com.fulcrumgenomics.FgBioDef.{FilePath, PathToBam}
+import com.fulcrumgenomics.commons.util.CaptureSystemStreams
+
+/** Integration tests for calling duplex consensus reads.
+  *
+  * Runs:
+  * - GroupReadsByUmi
+  * - CallDuplexConsensusReads
+  * - FilterConsensusReads
+  * */
+class DuplexConsensusCallingIntegrationTest extends UnitSpec with CaptureSystemStreams {
+
+  /** Make a reference file that is 100 lines of 100 As. */
+  private val ref = {
+    val builder = new ReferenceSetBuilder
+    builder.add("chr1").add("AAAAAAAAAA", 10 * 100) // 5000 bases
+    builder.toTempFile()
+  }
+
+  /** Creates a read pair where the bases are all As and sets the RX tag to "AAAAAAAA"*/
+  private def record(builder: SamBuilder, name: String, start1: Int, start2: Int, strand1: Strand=Plus, strand2: Strand=Minus, base: String = "A", umiBases: String = ""): Seq[SamRecord] = {
+    val rxValue = if (umiBases.nonEmpty) umiBases else {
+      if (strand1 == Plus) "ACG-GCT" else "GCT-ACG"
+    }
+    require(base.length == 1)
+    builder.addPair(name=name, start1=start1, start2=start2, strand1=strand1, strand2=strand2, attrs=Map("RX" -> rxValue)).map { rec =>
+      rec.bases = base*builder.readLength
+      rec
+    }
+  }
+
+  /** Create some various test cases. */
+  private val (inputBam, inputRecords) = {
+    val builder = new SamBuilder(readLength=100, sort=Some(SamOrder.Coordinate))
+
+    // group 0: one FR read pair -> duplex from one strand (if pass filters)
+    {
```

Comment:
Since none of these test cases create any local variables, I don't think there's a need to wrap them in `{}` blocks.